### PR TITLE
Update run_CIs_for_external_pr.py to skip passed checks

### PR DESCRIPTION
**Pull request by fs-eire**
### Description
Update run_CIs_for_external_pr.py to skip passed checks

### CODE REVIEW

**Review by skottmckay — APPROVED** [review]

File: tools/python/run_CIs_for_external_pr.py [hunk 1]
```diff
@@ -25,9 +25,9 @@ def parse_args():
     return args
 
 
-def run_gh_pr_command(command: typing.List[str]):
+def run_gh_pr_command(command: typing.List[str], check=True):
     try:
-        return subprocess.run(["gh", "pr", *command], capture_output=True, text=True, check=True)
+        return subprocess.run(["gh", "pr", *command], capture_output=True, text=True, check=check)
```
Comment (skottmckay) [inline tools/python/run_CIs_for_external_pr.py:30]: Out of interest why is check=False when checking output about passed pipelines? Do we expect that to fail for a valid PR that is currently open?

**Review by skottmckay — CHANGES_REQUESTED** [review]
Not sure this is working. Run against PR 16669 which is an externally submitted PR where the Azure pipelines have not been run yet. 
 
I tried manually and got this:

```
gh pr checks --required 16669
All checks were successful
0 failing, 2 successful, 0 skipped, and 0 pending checks

✓  Python format  4m24s  https://github.com/microsoft/onnxruntime/actions/runs/5537867454/job/15249917647
✓  license/cla           https://github.com/microsoft/contributorlicenseagreement
```

Which might indicate that the new logic requires the CIs to have been attempted once previously.

**Comment by skottmckay** [comment]
> Not sure this is working. 

Sorry - overlooked it is only removing CIs from the list so an empty list initially is fine. Would be good to capture that in some comments that it's expected the ORT CIs won't show up initially.

**Review by skottmckay — APPROVED** [review]

File: tools/python/run_CIs_for_external_pr.py (hunk 1 above)
Comment (fs-eire) [reply tools/python/run_CIs_for_external_pr.py:30]: `gh pr checks` exits with non-zero exit code when failures in pipeline exist

**Comment by fs-eire** [comment]
I tried multiple ways and cannot get the list for azp "required" pipelines before they starts.

**Comment by skottmckay** [comment]
> I tried multiple ways and cannot get the list for azp "required" pipelines before they starts.

That's really useful information that would be great to capture in a comment.

**Review by skottmckay — APPROVED** [review]

**Pull request merged**